mesonbuild/interpreterbase.py

```diff
@@ -926,6 +928,18 @@ def evaluate_ternary(self, node: mparser.TernaryNode) -> TYPE_var:
         else:
             return self.evaluate_statement(node.falseblock)
 
+    def evaluate_fstring(self, node: mparser.FormatStringNode) -> TYPE_var:
+        assert(isinstance(node, mparser.FormatStringNode))
+
+        def replace(match: T.Match[str]) -> str:
+            var = str(match.group(1))
+            try:
+                return str(self.variables[var])
+            except KeyError:
+                raise mesonlib.MesonException(f'Identifier "{var}" does not name a variable.')
+
+        return re.sub(r'{([_a-zA-Z][_0-9a-zA-Z]*)}', replace, node.value)
```

Comment:
In Python we can use {{ to escape, can we do it here too? Also I'm meson format was using @0@, I think we should continue to use that instead of curly brackets for consistency.

Reply:
This regex will replace {{x}} with {value} if x is currently defined. I don't think explicit escaping is required as using characters that do not constitute an identifier (such as whitespace) within the curlies will prevent substitution.
As for the consistency point, I deliberately went for the more familiar curly brackets instead of Meson's current notation because it looks clearer to me - are we sure we want to keep the old syntax?

Reply:
I chose not to use curly brackecs on purpose because they are used by so many other text replacement libraries as well as being a semantically meaningful character in most programming languages. The strings Meson expands often end up as inputs to other string replacements which means you have to do escaping and that gets nasty quite quickly.

Reply:
Alright, I'll adjust the syntax then.

Reply:
If accepted, it should behave the same as the existing `@0@` style placeholders, which I'll note do not do escaping:

```
message('@@0@@.format('hi'))
```

returns `Message: @hi@`